server/src/main/java/com/cloud/configuration/ConfigurationManagerImpl.java

```diff
@@ -2319,9 +2319,9 @@ public ServiceOffering createServiceOffering(final CreateServiceOfferingCmd cmd)
                 throw new InvalidParameterValueException("For creating a custom compute offering cpu and memory all should be null");
             }
             // if any of them is null, then all of them shoull be null
-            if (maxCPU == null || minCPU == null || maxMemory == null || minMemory == null) {
-                if (maxCPU != null || minCPU != null || maxMemory != null || minMemory != null) {
-                    throw new InvalidParameterValueException("For creating a custom compute offering min/max cpu and min/max memory should all be specified");
+            if (maxCPU == null || minCPU == null || maxMemory == null || minMemory == null || cpuSpeed == null) {
+                if (maxCPU != null || minCPU != null || maxMemory != null || minMemory != null || cpuSpeed != null) {
+                    throw new InvalidParameterValueException("For creating a custom compute offering min/max cpu and min/max memory/cpu speed should all be null or all specified");
                 }
             } else {
                 if (cpuSpeed != null && (cpuSpeed.intValue() < 0 || cpuSpeed.longValue() > Integer.MAX_VALUE)) {
```

Comment:
`cpuSpeed` null check not required here, it is always not null as per the above changes.

Reply:
@sureshanaparti indeed, I will remove it.